rgo/builders/copr.py

```diff
@@ -161,19 +164,44 @@ def build(self, component, with_build_id=None, after_build_id=None):
             "after_build_id": after_build_id,
         }
 
-        if self.chroots:
-            buildopts["chroots"] = self.chroots
-
-        build = self.client.build_proxy.create_from_file(self.owner, self.name, component.srpm, buildopts=buildopts)
-        LOGGER.info('Submitted Copr build for %s chroots: %s %s%sURL: %s' % (
-            component.name,
-            ", ".join(buildopts["chroots"]) if "chroots" in buildopts else "(all)",
-            "with: %s " % with_build_id if with_build_id is not None else "",
-            "after: %s " % after_build_id if after_build_id is not None else "",
-            self._build_url(build.id)
-        ))
+        if self.cmdline_input_chroots:
+            buildopts["chroots"] = self.cmdline_input_chroots
+
+        overriden_chroots = set()
+        for override in component.overrides:
+            overriden_chroots.update(override.roots)
+
+        # if we have atleast one default (not overriden) chroot do the default build
+        if (list(self.all_project_chroots - overriden_chroots)):
+            buildopts["chroots"] = list(self.all_project_chroots - overriden_chroots)
```

Comment:
This just doesn't do what it's supposed to. The logic behind the chroots is this:
- If you don't specify `--chroots` on CLI, it builds for all chroots of the Copr project
- If you specify `--chroots` on CLI, it builds the chroots you've specified (and adds them to the Copr project if they weren't there before)
- If you don't specify `--chroots` and the Copr project doesn't exist yet, it errors out saying you need to specify the chroots for the new project

Here you override the previously set `buildopts["chroots"]` unconditionally, always building all project chroots.

Reply:
I see, I have misunderstood that, I though the chroots from command line are just added. WIll fix it.